hack/sync-forks.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash -e
+
+# This script synchronizes all our forks with the UPSTREAM commits to appropriate repositories.
+#
+
+set +o errexit
+
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
+
+TMPDIR=$(mktemp -d)
+
+function sync_fork() {
+    local fromRepo="$1"
+    local fromBranch="$2"
+    local fromDir="$3"
+    local toRepo="$4"
+    local toBranch="$5"
+    local fromRepoName=$(basename ${fromRepo})
+    local toRepoName=$(basename ${toRepo})
+
+    fromRepoLocation="${TMPDIR}/${fromRepoName}"
+    toRepoLocation="${TMPDIR}/${toRepoName}"
+
+    mkdir -p "${fromRepoLocation}"
+    mkdir -p "${toRepoLocation}"
+
+    pushd "${fromRepoLocation}" >/dev/null
+    # if we already have a git repo, assume it's the right one and get the requested branch
+    git rev-parse --is-inside-work-tree || git clone --single-branch --branch=${fromBranch} -- "${fromRepo}" ${fromRepoLocation}
+    git fetch --refmap -- origin refs/heads/${fromBranch}:refs/remotes/origin/${fromBranch}
+    git checkout ${fromBranch}
+    git reset --hard origin/${fromBranch}
+    firstFromSHA=$(git log --reverse --oneline  --format='%H' | head -n 1)
+    newFromSHA=$(git log --oneline --format='%H' -1)
+    popd >/dev/null
+
+    pushd "${toRepoLocation}" >/dev/null
+    # if we aren't in a git repo, clone it
+    git rev-parse --is-inside-work-tree || git clone --single-branch --branch=${toBranch} -- "${toRepo}" ${toRepoLocation}
+    git fetch --refmap -- origin refs/heads/${toBranch}:refs/remotes/origin/${toBranch}
+    git checkout ${toBranch}
+    git reset --hard origin/${toBranch}
+    startingFromSHA=${firstFromSHA}
+    if [ -f "${fromRepoName}.sha" ]; then
+        startingFromSHA=$(cat ${fromRepoName}.sha)
+    fi
+    if [ "${newFromSHA}" == "${startingFromSHA}" ]; then
+        popd >/dev/null
+        echo "already at level: ${newFromSHA}, nothing to sync"
+        exit 0
+    fi
+    echo "syncing from ${startingFromSHA}..${newFromSHA}"
+    # create a clean branch to start from
+    git branch -D "${toBranch}-working" || true
+    git checkout -b "${toBranch}-working"
+    popd >/dev/null
+
+    pushd "${fromRepoLocation}" >/dev/null
+    patchDir="${TMPDIR}/patches"
+    rm -rf ${patchDir} && mkdir -p "${patchDir}"
+    index=0
+    for commit in $(git log --format='%H' --no-merges --reverse -- "${startingFromSHA}..${newFromSHA}" "${fromDir}"); do
```

Comment:
@deads2k it looks like the double hyphen before startingFromSHA is causing to get more commits than the range you specify.

Reply:
> @deads2k it looks like the double hyphen before startingFromSHA is causing to get more commits than the range you specify.

wrong spot I guess.  Move to before `fromDir`